liota/dccs/iotcc.py

```diff
@@ -157,7 +159,18 @@ def on_response(msg, reg_resp_q):
                     self._store_reg_entity_attributes("Devices", entity_obj.name, self.reg_entity_id,
                                                       entity_obj.entity_type, None)
 
-            return RegisteredEntity(entity_obj, self, self.reg_entity_id)
+            _reg_entity_obj = RegisteredEntity(entity_obj, self, self.reg_entity_id)
+            if self._sys_properties:
+                _sys_prop_dict = ast.literal_eval(self._sys_properties)
+                if isinstance(_sys_prop_dict, dict) and _sys_prop_dict:
```

Comment:
The second check is redundant as isinstance() takes care of None. Please remove it.

Reply:
The second condition checks if the passed dictionary is not empty, I have tested it 'isinstance()' doesn't take care of that.

Reply:
I tried out with the following code: 
#obj = {"a":"b"}
#obj = None
obj = {}
if isinstance(obj, dict) and obj:
  print "true"
else:
  print "false"

It printed true only when obj = {"a":"b"} and false for the other two. Am i missing something?

Reply:
Yea because the condition is "if isinstance(obj, dict) and obj:", you have the second check which is part of the current PR as well.

Reply:
Understood. Thanks for pointing it out.